### DISCUSSION TITLE
How do I compile multiple directories at the same time?

### DISCUSSION THREAD

**Discussion by claytoneast** [swc-project/swc#2815] [Q&A]
First off, apologies if I could have pieced together what might be an already-existing way to do this from the docs. I've been reading them and experimenting for 3 hours now, and I can't figure it out.

I have an app. The relevant pieces of directory structure of this app:
```
- src
	- someRoute.ts
- test
	- someTest.ts
- tools
	- someTool.ts
	- anotherToolButThisOneDoesntGetCompiledOrSentToProd.js
- dist
.swcrc
```

We use typescript, and thus `tsc`. In our `tsconfig.json`, we have (relevant opts only here)
```
"compilerOptions": {
  "outDir": "dist",
},
"include": [
  "src/**/*",
  "test/**/*",
  "tools/someTool.ts",
]
```

This allows us to run `tsc`, and get all the things in our `include` option paths compiled into the `dist` directory.

I'm wondering if `swc` has a good way to do this. I have a really bad way to do it:
```
# .swcrc
{
  "test": [
    "tools",
    "src",
    "test"
  ],
  "exclude": [
    "node_modules",
    "dist"
  ]
}
```
in conjunction with `$ ./node_modules/.bin/swc ./* -d dist/src --config-file .swcrc`

It's bad because while `swc` only takes between 300ms and 900ms to compile, the entire command takes \~5s. I'm guessing it takes this long because I'm setting the input directory as the entire app folder, and so `swc` has to get every file within the app, test it against the "test" and "exclude" patterns,  etc (thanks `node_modules` w/ 43k files).

Another terrible way that I thought of (but haven't tried) would be to start 3 separate `swc` processes, each pointed at the respective directory I want.

```
$ ./node_modules/.bin/swc src -d dist/src --config-file .swcrc
$ ./node_modules/.bin/swc test -d dist/test --config-file .swcrc
$ ./node_modules/.bin/swc tools -d dist/tools --config-file .swcrc
```

I don't really want to do that^ option either. 

It seems like there should be a way to have an `include` option, or a `srcDirs` option, or something that lets me do this without scanning every single filename in my app directory. Would be really nice. It is the only thing preventing us from switching right now, and boy would it be awesome to switch, the compile speed is like playing a game at 10fps, upgrading your computer, and having it run at 120fps. It's so nice.

Please let me know if there is some config option that does this (that I am apparently incapable of reading and understanding), or if this isn't supported yet.

Thanks!!

**Comment by bahtou** [comment]
any updates on this?

**Comment by akwodkiewicz** [comment]
I have created this issue: https://github.com/swc-project/swc/issues/8139
If it was solved, I guess you would then be able to run `swc . -d dist` and compile everything defined in `.swcrc` with a single command.

**Reply by ncjones** [reply]
This looks like what I need. But what do you mean by "compile everything defined in .swcrc"? Is there an "include" option? From what I found so far there is only an "exclude" option.

**Reply by akwodkiewicz** [reply]
You are right, "compile everything defined in .swcrc" does not make sense, I thought it has a setting similar to `tsconfig.json`'s "include". Disregard this part 😅 

What I was trying to say is just that for me it is now enough to transpile `.`, because the root project directory contains all of the folders with TS files. Seems to be the case for you as well (`src`, `test`, `tools`).

**Reply by akwodkiewicz** [reply]
And BTW, it has been resolved 4 days ago already (see the issue)! Don't know if the fix is already available in some release or not, though.
